iree/hal/vulkan/api.h

```diff
@@ -89,12 +96,66 @@ iree_hal_vulkan_syms_release(iree_hal_vulkan_syms_t* syms);
 
 #endif  // IREE_API_NO_PROTOTYPES
 
+//===----------------------------------------------------------------------===//
+// iree::hal::vulkan Extensibility Util
+//===----------------------------------------------------------------------===//
+
+#ifndef IREE_API_NO_PROTOTYPES
+
+// Gets the names of the Vulkan instance extensions needed to create a driver
+// with |iree_hal_vulkan_driver_create_using_instance| and |options|.
+//
+// |out_count| is a pointer to an integer related to the number of extensions.
+// |out_extensions| is either nullptr or a pointer to an array to be filled with
+// the required Vulkan driver extensions.
+IREE_API_EXPORT iree_status_t IREE_API_CALL
```

Comment:
prefer a style like this:
https://source.corp.google.com/piper///depot/google3/third_party/iree/iree/rt/api.h;l=381

This makes it explicit which are input args defining the array bounds (`result_capacity`), output args for queries (`out_result_count`), and the list itself. The error handling code also takes care of verifying the sizes.